### PR TITLE
Add otlp exemplar ingestion test

### LINKED ISSUE #6235: Add exemplars coverage in OTLP integration test
**Is your feature request related to a problem? Please describe.**

The existing OTLP push integration test covers the ingestion of both series and histograms. However, we didn't cover exemplar data type.
https://github.com/cortexproject/cortex/blob/master/integration/otlp_test.go 

**Describe the solution you'd like**

Ingest exemplars data in the integration test and make sure the exemplars can be ingested successfully via OTLP endpoint.

### PR DESCRIPTION
**Pull request by SungJin1212**
Add an otlp exemplar ingestion test.

**Which issue(s) this PR fixes**:
Fixes #6235 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by SungJin1212** [comment]
@yeya24 
I think metadata ingestion test is needed as well. Let me add it.

**Comment by yeya24** [comment]
@SungJin1212 I don't think OTLP protocol supports metadata as metadata is only a Prometheus thing

**Comment by SungJin1212** [comment]
@yeya24 
I just found out the otlp ingestion of the metadata is not implemented yet. Does that mean that the implementation itself is not necessary?

**Comment by yeya24** [comment]
https://github.com/prometheus/prometheus/blob/main/storage/remote/write_handler.go#L527
If you look at the Prometheus OTLP handler implementation it only ingests timeseries.

https://opentelemetry.io/docs/specs/otlp/ 
Metadata is only part of Prometheus remote write protocol but not part of OTLP specification

**Comment by SungJin1212** [comment]
@yeya24 
I got it. Thank you!

**Comment by yeya24** [comment]
https://github.com/prometheus/prometheus/pull/13439 @SungJin1212 We can cover metadata API after this upstream PR gets merged

File: integration/e2ecortex/client.go [hunk 1]
```diff
@@ -231,6 +231,73 @@ func convertTimeseriesToMetrics(timeseries []prompb.TimeSeries) pmetric.Metrics
...
+	}
+
+	// Create HTTP request
+	req, err := http.NewRequest("POST", fmt.Sprintf("http://%s/api/v1/otlp/v1/metrics", c.distributorAddress), bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("X-Scope-OrgID", c.orgID)
+	req.Header.Set("Content-Type", "application/x-protobuf")
```
Comment (yeya24) [inline integration/e2ecortex/client.go:285]: Just curious, is this header required?

File: integration/e2ecortex/client.go [hunk 2]
```diff
@@ -231,6 +231,73 @@ func convertTimeseriesToMetrics(timeseries []prompb.TimeSeries) pmetric.Metrics
...
+
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
```
Comment (yeya24) [inline integration/e2ecortex/client.go:288]: I suggest we pass ctx from the test itself so we control timeout there

File: integration/e2ecortex/client.go [hunk 3]
```diff
@@ -267,6 +334,13 @@ func (c *Client) Query(query string, ts time.Time) (model.Value, error) {
 	return value, err
 }
 
+// QueryExemplars runs an exemplars query
+func (c *Client) QueryExemplars(query string, start, end time.Time) ([]promv1.ExemplarQueryResult, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
```
Comment (yeya24) [inline integration/e2ecortex/client.go:340]: I suggest we pass ctx from the test itself so we control timeout there

**Review by yeya24 — COMMENTED** [review]
Thanks. Great work. Just some nits

File: integration/e2ecortex/client.go (hunk 1 above)
Comment (SungJin1212) [reply integration/e2ecortex/client.go:285]: If `Content-Type" is empty, we got the err msg
`level=error org_id=fake err="unsupported content type: , supported: [application/json, application/x-protobuf]"`

But `X-Scope-OrgID` is not required  (`auth_enabled: false`)

File: integration/e2ecortex/client.go (hunk 3 above)
Comment (SungJin1212) [reply integration/e2ecortex/client.go:340]: It would be better but we cannot directly access the client's timeout in the test.

Comment (yeya24) [reply integration/e2ecortex/client.go:340]: I see nvm it was the same pattern for other APIs. I am good

File: integration/e2ecortex/client.go (hunk 1 above)
Comment (yeya24) [reply integration/e2ecortex/client.go:285]: Thanks for pointing it out

**Review by yeya24 — APPROVED** [review]

**Pull request merged**